### PR TITLE
Blog post: motivating questions for a Carpentries-style workshop on social sciences for programmers

### PR DESCRIPTION
**Pull request by gvwilson**
Please review inline, add comments below, or [mail me](mailto:gvwilson@third-bit.com).

### CODE REVIEW

File: _posts/2024/2024-01-15-the-first-five-questions.md [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+    but that makes it difficult or impossible to compare people.
+
+Who owns the software I write?
+:   Depending on your circumstances,
+    the software you write may belong to your employer
+    even if you wrote it on your own time using your own computer
+    (because they provided the training you needed to write it),
+    or they may only have an interest in it if you used their physical resources,
+    or it may be entirely yours as long as you wrote it on your own time.
+    The answer depends on your venue's rules about intellectual property;
```
Comment (yabellini) [inline _posts/2024/2024-01-15-the-first-five-questions.md]: What you say venue’s rules what do you mean? 

At least in Argentina there is a intellectual property law that rules who owns the software. For example, if you program you are always the author but who have the rights is different if you are and employee or not.

File: _posts/2024/2024-01-15-the-first-five-questions.md [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+    that concept is less than two hundred years old,
+    but has expanded steadily over the last century
+    via processes such as regulatory capture
+    in ways that further the interests of profit-driven corporations.
+
+Why do hairdressers have to be licensed but programmers don't?
```
Comment (yabellini) [inline _posts/2024/2024-01-15-the-first-five-questions.md]: I am not sure if this is the case all over the world. For example, I have what we call here a professional registration. I have to register with the Colegio de ingenieros, licenciados y técnicos de la pampa and pay a fee every year. Without that I cannot practice. But that is not mandatory in every province. 
Your point is still valid. Why you choose the hairdresser as example? I can think also in truck or taxi drivers as another example that perhaps is a more world wide case.

File: _posts/2024/2024-01-15-the-first-five-questions.md [hunk 3]
```diff
@@ -0,0 +1,110 @@
+---
+title: "The First Five Questions"
+date: 2024-01-15
+year: 2024
+---
+
+I've been talking and writing for several years
+about the need for something that would teach programmers
+the useful basics of economics, politics, sociology, and other social sciences.
+A few days ago I framed it as [the Scratch of social sciences][like-scratch];
```
Comment (csl29) [inline _posts/2024/2024-01-15-the-first-five-questions.md]: OH what an interesting question. Tbh I think because the social sciences are the science of our human lives, the scratch is actually critically reflecting on our lives through a science approach  - which both muddies and clarifies this. I've used Scratch maybe once, but my understanding is that Scratch succeeds in teaching CS by giving you a project that requires it in a friendly, fun, and relevant way. It doesn't teach you The Language of All Languages or The Rules of All Languages, but it teaches you the processes and outlook needed. So if I think about that and apply it to how IRL professors teach social sciences, I think about how we teach people how to observe and ask questions about life experiences - adding the "how" and "why" questions to observations, then teaching them to answer those questions by linking them back to social science theories and research. Teaching them how to science, so to speak.

File: _posts/2024/2024-01-15-the-first-five-questions.md [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+in the way that Scratch does.
+
+One possibility is something like the [Carpentries][carpentries] workshops.
+A typical workshop spends half a day on the Unix shell,
+half a day on version control,
+and a full day on either Python or R;
+all of the lessons are aimed at people who have little or no previous training in programming
+(though they may often have picked up bits and pieces in a data analysis class).
+
+Could that format be used to teach a few actionable fundamentals of the social sciences?
```
Comment (csl29) [inline _posts/2024/2024-01-15-the-first-five-questions.md]: I think this is an excellent approach/idea and one that is consistent with the approach I noted by profs as a practical "what do you do if its a large group of people and not a special seminar series where everyone has time to get your feedback and help," where you define common experiences and the how and why questions associated with them, and walk through the process together.

File: _posts/2024/2024-01-15-the-first-five-questions.md [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+    in ways that further the interests of profit-driven corporations.
+
+Why do hairdressers have to be licensed but programmers don't?
+:   First,
+    notice how the question uses the social status of the two professions:
+    it could be rewritten as,
+    "Isn't it ridiculous that lowly hairdressers have to get licenses
+    but exalted programmers don't?"
+    The answer is that licensure is tied to professional responsibility
+    and to the way in which middle-class professions such as law and medicine
```
Comment (csl29) [inline _posts/2024/2024-01-15-the-first-five-questions.md]: although law and medicine do require licensure and continuing education training standards. So I wonder if the question here is actually more about "Why don't programmers globally require licensure when so many other professions do?"

File: _posts/2024/2024-01-15-the-first-five-questions.md [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+
+Why do we have so many meetings?
+:   Coordination costs rise as the square of the number of people,
+    so large groups must (a)
+    make work uniform
+    so that sets of people are effectively interchangeable (e.g., assembly lines and typing pools),
+    (b) create sub-groups and invest effort in coordinating their work,
+    or (c) allow sub-groups to work independently and pay the cost of duplicated or contradictory effort.
+
+Why do so many meetings suck?
```
Comment (csl29) [inline _posts/2024/2024-01-15-the-first-five-questions.md]: I think to make this broader meetings topic more actionable, a question could be "How do I make meetings not suck?" and combine the two topics together to address the function of meetings, pitfalls of crappy meetings, and tips for holding a good meeting according to social sciences

**Review by klbarnes20 — COMMENTED** [review]
I think this is a really interesting idea. As a social psychologist (but a newish programmer) I can see a million potential things that could be useful to teach programmers, but I don't think I know enough about what programmers are like. I think I would want to know more about the gaps you see in programmers' knowledge that something like this could fill in order to think more about those motivating questions. But, off the top of my head, social dynamics such as power (as you suggested), how to influence or persuade, groupthink, cognitive biases (e.g., availability heuristic, confirmation bias, etc.), the way social science and humanities approach problems.

File: _posts/2024/2024-01-15-the-first-five-questions.md (hunk 3 above)
Comment (dhavide) [reply _posts/2024/2024-01-15-the-first-five-questions.md]: I'm in agreement that a foundation in social sciences is something most programmers would benefit from, but it's not obvious to me that computer scientists in industry or academia would agree. So the opening question "What could we teach in a one-day workshop?" in some sense also needs to address the question "With whom can we get buy-in to get their team of devs to do this workshop?" I can't see anyone in industry doing this. Perhaps academics might, but they may want to see how this gets them closer to publishing something.

File: _posts/2024/2024-01-15-the-first-five-questions.md (hunk 1 above)
Comment (dhavide) [reply _posts/2024/2024-01-15-the-first-five-questions.md]: The question "Who owns the software I write?" is one that, in my mind, leads to questions about ethics or law more than social sciences. This question perhaps offers a way to get buy-in from CS faculty more than the others.

**Review by dhavide — COMMENTED** [review]
Another question to consider:

> In 1960, X% of programmers in the US were women as compared to Y% by the end of the century. What factors lead to this demographic shift and what does they say about the tech industry today?

I don't know the actual numbers and there well may be cleverer ways to pose this question. But it is one that, I believe, points to questions about social systems and how they operate. This question can be a minefield, I know, but it seems aligned with the social themes we want to consider.

File: _posts/2024/2024-01-15-the-first-five-questions.md (hunk 1 above)
Comment (gvwilson) [reply _posts/2024/2024-01-15-the-first-five-questions.md]: I was thinking about aiming this at working programmers rather than at CS faculty; I've added a note about Frank (prototypical audience member) that I hope clarifies that.

File: _posts/2024/2024-01-15-the-first-five-questions.md (hunk 2 above)
Comment (gvwilson) [reply _posts/2024/2024-01-15-the-first-five-questions.md]: I think any profession that is (a) licensed and (b) seen as having lower social standing than programming could work here.

File: _posts/2024/2024-01-15-the-first-five-questions.md (hunk 3 above)
Comment (gvwilson) [reply _posts/2024/2024-01-15-the-first-five-questions.md]: My working title is "What Silicon Valley Doesn't Want You to Know About How Politicis, Economics, Justice, and Power Actually Work", but that feels a bit wordy :-)
